docs/topics/upgrade.md

```diff
@@ -218,3 +218,16 @@ You *may not* change the following values, doing so may break your cluster!
 - DO NOT CHANGE the static IP address range of your control plane via `masterProfile.firstConsecutiveStaticIP`
 
 These types of configuration changes are advanced, only do this if you're a confident, expert Kubernetes cluster administrator!
+
+### How can I recreate upgraded nodes that did not reach the `Ready` state?
+
+If any of the upgraded nodes is not able to reach the `Ready` state, then `aks-engine upgrade` will exit showing a message similar to one below:
+
+```console
+Error validating upgraded master VM: k8s-master-12345678-0
+Error: upgrading cluster: Node was not ready within 20m0s
+```
+
+These errors are typically caused by bad or incompatible `properties.orchestratorProfile.kubernetesConfig` property values in your API model that prevent `kubelet` (or control plane components) from starting successfully.
```

Comment:
What makes us think this is the most common cause of kubelet failing to come online? If we have definitive proof of these errors, we should be able to validate known incompatibilites during template generation/validation.

Reply:
The instance when I have seen this happening was upgrading ASH clusters from old AKS Engine versions, the issue was an incorrect pause image. That issue is fixed now by this #3922. 

That statement is meant to hint users to check around that area if nodes are not ready after upgrade. Let me know how do you prefer to document this.

Reply:
I would simply suggest "There are a variety of reasons why your cluster might not be able to come back online after an upgrade. Looking at the kubelet logs (`sudo journalctl -u kubelet`) may be a good first investigative step."

Or something like that.